src/Illuminate/Validation/Validator.php

```diff
@@ -1091,6 +1091,20 @@ public function getRules()
         return $this->rules;
     }
 
+    /**
+     * Get the validation rules with key placeholders removed.
+     *
+     * @return array
+     */
+    public function getRulesWithoutPlaceholders()
+    {
+        return collect($this->rules)
+            ->mapWithKeys(fn ($value, $key) => [
+                str_replace($this->dotPlaceholder, '\\.', $key) => $value,
```

Comment:
This is not using the existing method for replacing placeholders, as it does not restore the backslash.